x/emissions/types/validations.go

```diff
@@ -663,52 +663,61 @@ func (msg *RemoveRegistrationRequest) Validate() error {
 }
 
 // helper for stake validation functions
-func stakeValidateHelper(addr []string, amount cosmosMath.Int) error {
+func stakeValidateHelper(addr []string, amount cosmosMath.Int, allowZeroAmount bool) error {
 	for _, ad := range addr {
 		_, err := sdk.AccAddressFromBech32(ad)
 		if err != nil {
 			return errors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", ad)
 		}
 	}
-	if amount.LT(cosmosMath.ZeroInt()) {
-		return errors.Wrapf(sdkerrors.ErrInvalidCoins, "invalid amount (%s)", amount.String())
+	if amount.IsNegative() {
+		return errors.Wrapf(sdkerrors.ErrInvalidCoins, "amount must be non-negative: %s", amount.String())
 	}
+
+	if !allowZeroAmount && amount.IsZero() {
+		return errors.Wrapf(sdkerrors.ErrInvalidCoins, "amount must be positive: %s", amount.String())
+	}
+
 	return nil
 }
 
 // validate that an add stake request follows the expected format
 func (msg *AddStakeRequest) Validate() error {
-	return stakeValidateHelper([]string{msg.Sender}, msg.Amount)
+	return stakeValidateHelper([]string{msg.Sender}, msg.Amount, false)
 }
 
 // validate that a remove stake request follows the expected format
 func (msg *RemoveStakeRequest) Validate() error {
-	return stakeValidateHelper([]string{msg.Sender}, msg.Amount)
+	return stakeValidateHelper([]string{msg.Sender}, msg.Amount, false)
 }
 
 // validate that a delegate stake request follows the expected format
 func (msg *DelegateStakeRequest) Validate() error {
-	return stakeValidateHelper([]string{msg.Sender, msg.Reputer}, msg.Amount)
+	if msg.Reputer == msg.Sender {
+		return errors.Wrap(sdkerrors.ErrInvalidRequest, "cannot self delegate")
+	}
+
+	return stakeValidateHelper([]string{msg.Sender, msg.Reputer}, msg.Amount, false)
 }
 
 // validate that a remove delegate stake request follows the expected format
 func (msg *RemoveDelegateStakeRequest) Validate() error {
-	return stakeValidateHelper([]string{msg.Sender, msg.Reputer}, msg.Amount)
+	return stakeValidateHelper([]string{msg.Sender, msg.Reputer}, msg.Amount, false)
```

Comment:
Can you leave some comments on this and the following methods explaining the boolean?  i.e. Why do these various request types allow or disallow zero amounts